src/main/java/tech/picnic/jolo/Entity.java

```diff
@@ -61,6 +60,31 @@ private Entity(Table<R> table, Class<T> type, Field<Long> primaryKey, Field<?>[]
     this.fields = fields;
   }
 
+  @Override
+  public boolean equals(@Nullable Object o) {
```

Comment:
Why are we doing this? Two distinct Entity objects should never be considered equal IMO. You can create two different Entity objects for the same table, and use both to load the result of a single query (e.g. a self join).

Reply:
Ah yes, sorry. You'd of course use 2 different aliases for the tables. Then it makes sense. Still not sure why we're adding a definition for equals though, just because we can?

Reply:
It felt natural to override `equals` such that it is in sync with `hashCode` but it is indeed not necessary.
API-wise what is better?

Reply:
I think I would not override `equals`/`hashCode`, simply because I cannot think of a natural use case for wanting to have 2 instances of this class that should be considered equal.

Reply:
(But this is a highly subjective of course, so also happy to be convinced otherwise)

Reply:
The idea was to guard against accidental misuse since 2 entities that refer to the same table and same fields must effectively be equivalent, otherwise the corresponding SQL query would be invalid.
In that case the loader would automatically squash these entities into one.

It's indeed an unlikely scenario, though (guessing here since I haven't seen Jolo examples outside of Picnic). Adding this API characteristic would also constrain further API evolution and possibly prevent some unforeseen use cases.

Another option would be to emit a warning log if aforementioned misuse is detected during loader-building.

Reply:
Nevermind, there's already a check in the builder that states that `"Distinct entities cannot refer to the same primary key"`.

Reply:
I'd leave it out then. The default semantics meet the requirements for Jolo, and the most obvious form of abuse is guarded by a fail fast mechanism. If use cases do pop up, then we'll have some real requirements to work with and we can open an new PR based on those.